pinot-broker/src/test/java/org/apache/pinot/broker/broker/HelixBrokerStarterTest.java

```diff
@@ -18,287 +18,197 @@
 
-public class HelixBrokerStarterTest extends ControllerTest {
-  private static final int SEGMENT_COUNT = 6;
-  private static final String RAW_DINING_TABLE_NAME = "dining";
-  private static final String DINING_TABLE_NAME = TableNameBuilder.OFFLINE.tableNameWithType(RAW_DINING_TABLE_NAME);
-  private static final String COFFEE_TABLE_NAME = TableNameBuilder.OFFLINE.tableNameWithType("coffee");
 
-  private final Configuration _brokerConf = new BaseConfiguration();
+public class HelixBrokerStarterTest extends ControllerTest {
+  private static final String RAW_TABLE_NAME = "testTable";
+  private static final String OFFLINE_TABLE_NAME = TableNameBuilder.OFFLINE.tableNameWithType(RAW_TABLE_NAME);
+  private static final String REALTIME_TABLE_NAME = TableNameBuilder.REALTIME.tableNameWithType(RAW_TABLE_NAME);
+  private static final String TIME_COLUMN_NAME = "daysSinceEpoch";
+  private static final int NUM_BROKERS = 3;
+  private static final int NUM_SERVERS = 1;
+  private static final int NUM_OFFLINE_SEGMENTS = 5;
 
-  private ZkClient _zkClient;
-  private HelixBrokerStarter _helixBrokerStarter;
-  private ZkStarter.ZookeeperInstance _zookeeperInstance;
+  private HelixBrokerStarter _brokerStarter;
 
-  @BeforeTest
+  @BeforeClass
   public void setUp()
       throws Exception {
-    _zookeeperInstance = ZkStarter.startLocalZkServer();
-    _zkClient = new ZkClient(ZkStarter.DEFAULT_ZK_STR);
-
+    startZk();
     startController();
 
-    _brokerConf.addProperty(CommonConstants.Helix.KEY_OF_BROKER_QUERY_PORT, 8943);
-    _brokerConf.addProperty(CommonConstants.Broker.CONFIG_OF_BROKER_REFRESH_TIMEBOUNDARY_INFO_SLEEP_INTERVAL, 100L);
-    _helixBrokerStarter = new HelixBrokerStarter(_brokerConf, getHelixClusterName(), ZkStarter.DEFAULT_ZK_STR);
-    _helixBrokerStarter.start();
+    Configuration brokerConf = new BaseConfiguration();
+    brokerConf.addProperty(Helix.KEY_OF_BROKER_QUERY_PORT, 18099);
+    brokerConf.addProperty(Broker.CONFIG_OF_BROKER_REFRESH_TIMEBOUNDARY_INFO_SLEEP_INTERVAL, 100L);
+    _brokerStarter = new HelixBrokerStarter(brokerConf, getHelixClusterName(), ZkStarter.DEFAULT_ZK_STR);
+    _brokerStarter.start();
 
     ControllerRequestBuilderUtil
-        .addFakeBrokerInstancesToAutoJoinHelixCluster(getHelixClusterName(), ZkStarter.DEFAULT_ZK_STR, 5, true);
+        .addFakeBrokerInstancesToAutoJoinHelixCluster(getHelixClusterName(), ZkStarter.DEFAULT_ZK_STR, NUM_BROKERS - 1,
+            true);
     ControllerRequestBuilderUtil
-        .addFakeDataInstancesToAutoJoinHelixCluster(getHelixClusterName(), ZkStarter.DEFAULT_ZK_STR, 1, true);
-
-    while (_helixAdmin.getInstancesInClusterWithTag(getHelixClusterName(), "DefaultTenant_OFFLINE").size() == 0
-        || _helixAdmin.getInstancesInClusterWithTag(getHelixClusterName(), "DefaultTenant_BROKER").size() == 0) {
-      Thread.sleep(100);
-    }
+        .addFakeDataInstancesToAutoJoinHelixCluster(getHelixClusterName(), ZkStarter.DEFAULT_ZK_STR, NUM_SERVERS, true);
```

Comment:
[since you are doing a bunch of cleanup] can you please change this to addFakeServerInstancesToHoinHelixCluster() ?

Reply:
There are too many usages for this method, so I would do the name name in another pr.